src/Symfony/Component/Security/Http/LoginLink/LoginLinkHandler.php

```diff
@@ -90,12 +95,15 @@ public function consumeLoginLink(Request $request): UserInterface
             throw new InvalidLoginLinkException('Missing "expires" parameter.');
         }
 
+        $parameters = $request->query->all();
```

Comment:
This is impossible without breaking many code live, because
* webmails can append query parameters
* ...entreprise proxies/network too


That could break authentification for every link out there that has been generated

Reply:
I've never seen actual attributes added to the query, but I assume it could happen. 

I'll update the link so that it contains the list of keys used to generate the signature, and will only extract those keys from that parameter bag, ignoring all others.

Reply:
https://mailchimp.com/resources/utm-medium-email/

https://emaillabs.io/en/cloud-smtp-integration-with-google-analytics/

https://www.campaignmonitor.com/blog/email-marketing/utm-codes-in-email/

;)

Reply:
You'd send an e-mail with pre-validated authentication credential through a newsletter delivery service? 

Wherever I came by any kind of tracking, it was done by the service replacing all links in the email to make them point at a redirecting proxy/tracker, which then forwarded the request to the original URL.

In any case, I don't disagree. Accepting all query parameters isn't a good idea, period.